code/weapon/weapons.cpp

```diff
@@ -7534,6 +7512,33 @@ void validate_SSM_entries()
 	}
 }
 
+int weapon_get_random_usable_weapon()
+{
+	int rand_wep;
+	int idx = 0, weapon_list[MAX_WEAPON_TYPES];
```

Comment:
Maybe use a vector for `weapon_list` to reduce memory consumption and have compatibility for a future dynamic weapons limit.

Reply:
Ok.